test/lib/tasks/generateVersionInfo.js

```diff
@@ -156,3 +182,784 @@ test("integration: Library without i18n bundle file failure", async (t) => {
+		string: `
+			<?xml version="1.0" encoding="UTF-8" ?>
+			<library xmlns="http://www.sap.com/sap.ui.library.xsd" >
+				<name>${names.join(".")}</name>
+				<vendor>SAP SE</vendor>
+				<copyright></copyright>
+				<version>2.0.0</version>
+
+				<documentation>Library ${names.slice(1).join(".").toUpperCase()}</documentation>
+			</library>
+		`,
+		project: createProjectMetadata(names)
+	}));
+}
+
+/**
+ *
+ * @param {module:@ui5/fs.DuplexCollection} dependencies
+ * @param {module:@ui5/fs.resourceFactory} resourceFactory
+ * @param {string[]} names e.g. ["lib", "a"]
+ * @param {object[]} deps
+ * @param {string[]} [embeds]
+ */
+const createResources = async (dependencies, resourceFactory, names, deps, embeds) => {
+	await createDotLibrary(dependencies, resourceFactory, names);
+	await createManifestResource(dependencies, resourceFactory, names, deps, embeds);
+};
+
+
+test.serial("integration: Library with dependencies and subcomponent", async (t) => {
+	const workspace = createWorkspace();
+
+	await createDotLibrary(workspace, resourceFactory, ["test", "lib"]);
+
+	// input
+	// lib.a => lib.c, lib.b
+	// lib.b => lib.c (true)
+	// lib.c => lib.d
+	// lib.d => lib.e
+	// lib.e =>
+	// lib.a.sub.fold => lib.c
+
+	// expected outcome
+	// lib.a => lib.c, lib.b, lib.d
+	// lib.b => lib.c (true), lib.d (true)
+	// lib.c => lib.d
+	// lib.d => lib.e
+	// lib.e =>
+	// lib.a.sub.fold => lib.c, lib.d, lib.e
+
+	// dependencies
+	const dependencies = createDependencies({virBasePath: "/"});
+
+	// lib.a
+	const embeds = ["sub/fold"];
+	await createResources(dependencies, resourceFactory, ["lib", "a"], [{name: "lib.b"}, {name: "lib.c"}], embeds);
+	// sub
+	await createManifestResource(dependencies, resourceFactory, ["lib", "a", "sub", "fold"], [{name: "lib.c"}]);
+
+	// lib.b
+	await createResources(dependencies, resourceFactory, ["lib", "b"], [{name: "lib.c", lazy: true}]);
+
+	// lib.c
+	await createResources(dependencies, resourceFactory, ["lib", "c"], [{name: "lib.d"}]);
+
+	// lib.d
+	await createResources(dependencies, resourceFactory, ["lib", "d"], [{name: "lib.e", lazy: true}]);
+
+	// lib.e
+	await createResources(dependencies, resourceFactory, ["lib", "e"], []);
+
+	const oOptions = {
+		options: {
+			projectName: "Test Lib",
+			pattern: "/resources/**/.library",
+			rootProject: {
+				metadata: {
+					name: "myname"
+				},
+				version: "1.33.7"
+			}
+		},
+		workspace,
+		dependencies
+	};
+	await assertCreatedVersionInfo(t, {
+		"components": {
+			"lib.a.sub.fold": {
+				"library": "lib.a",
+				"manifestHints": {
+					"dependencies": {
+						"libs": {
+							"lib.c": {},
+							"lib.d": {},
+							"lib.e": {
+								"lazy": true
+							}
+						}
+					}
+				}
+			}
+		},
+		"libraries": [{
+			"manifestHints": {
+				"dependencies": {
+					"libs": {
+						"lib.b": {},
+						"lib.c": {},
+						"lib.d": {},
+						"lib.e": {
+							"lazy": true
+						}
+					}
+				}
+			},
+			"name": "lib.a",
```

Comment:
would you mind to make the name the first property? That makes it easier to understand.

Reply:
tackled with 874b160